packages/core/src/tracing/instrumentation/database/elasticsearchModern.js

```diff
@@ -30,8 +31,9 @@ exports.init = function init() {
   requireHook.onModuleLoad('@elastic/elasticsearch', instrument);
 };
 
-function instrument(es) {
-  const ESAPI = require('@elastic/elasticsearch/api');
+function instrument(es, esModuleFilename) {
+  const pathToEsApi = path.join(esModuleFilename, '..', 'api');
+  const ESAPI = require(pathToEsApi);
```

Comment:
This change solves the problem at hand. However, we could run into similar issues in the future, it is a non-obvious trap to fall into. I wonder how we can steer our future selves in the right direction to not simply `require` additional modules from the instrumented packages, but instead construct the path based on the filename the requireHook returns.

Thoughts?

Reply:
I would think that requiring a package into the customer's code by itself sounds like a yellow flag already, but I wouldn't be able to predict the impacts of doing it.

When it comes to potential preventions, I don't have any concrete idea at this point, but a combination of a custom lint rule + unit test + having a special function in our API to require modules inside instrumentations seems like an interesting path.

For instance, let's say we have a function `sensistiveRequire` or whatever name fits, we could at least have a try/catch wrap around it, which would scream at us when potential errors happen, so we could more easily debug and fix customer problems.

The next step would be to create a custom lint rule (which I honestly know little about it, and I am not even sure if this is actually possible) that would shout at us if we use `require` directly into instrumentations. Again, I am not sure if this is possible, as we DO require normally anyway, so Idk how we could differentiate one from the other.

Last but not least, try to write some test that evaluates the require somehow. Either by providing a different path for node_modules or something else.

Sorry for not providing a more concrete idea, but maybe this is some useful brainstorming starting point...

Reply:
The special function for such a require and the custom lint rule sound like good ideas. 👍

Reply:
I've extracted the require to a function in tracingUtil now and also added an eslint rule in a separate commit.